### PR TITLE
feat(nav): Use nav link if product docs only has one item

### PR DESCRIPTION
**Pull request by pbortnick**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-pbnav-dropdown-update-hashicorp.vercel.app/hcp) 🔎
- [Asana task](https://app.asana.com/0/1202097197789424/1203038566375268/f) 🎟️

## 🗒️ What

Update nav to use a link rather than a dropdown for Documentation if docs only has one nested item

## 🛠️ How

Two different approaches taken:
1. In the `<ProductPageHeaderContent />` component, check if the navItem contains more than one item and render a link or a dropdown. See [this commit](https://github.com/hashicorp/dev-portal/commit/ae521adc461e210ab5ab4dcfaa4bde6832615d87)
2. Scoped to docs only: in `getNavItems()` apply the check to the productData and create the documentation nav item obj based off that. (current approach)

This ticket seems to be scoped to documentation nav item only which is why I took the second approach. This also allows us to potentially have a nav item that has a dropdown with only one sub-item if we want in the future.

## 🧪 Testing

- [ ] Only the documentation nav item in hcp should have changed.
- [ ] Documentation link in hcp nav should go to /hcp/docs

### CODE REVIEW

File: src/lib/docs/get-docs-nav-items.ts [hunk 1]
```diff
@@ -5,6 +5,18 @@ import {
 	RootDocsPath,
 } from 'types/products'
 
+/**
+ * Check if docs has more than one sub-item
+ * to determine whether or not to show dropdown in nav
+ */
+
+export function checkDocsNavHasItems(
```
Comment (kendallstrautman) [inline src/lib/docs/get-docs-nav-items.ts:13]:
```suggestion
export function getDocsNavHasItems(
```
⛏️

File: src/lib/docs/get-docs-nav-items.ts [hunk 2]
```diff
@@ -5,6 +5,18 @@ import {
...
+	currentProduct: Pick<ProductData, 'rootDocsPaths' | 'docsNavItems'>
+): boolean {
+	const { docsNavItems, rootDocsPaths } = currentProduct
+	return !(!docsNavItems && rootDocsPaths.length === 1)
```
Comment (kendallstrautman) [inline src/lib/docs/get-docs-nav-items.ts:17]:
```suggestion
	return Boolean(docsNavItems) && rootDocsPaths.length > 1
```
💭 small thought: to make a logic a bit easier to read

**Review by kendallstrautman — APPROVED** [review]
LGTM! Just a few nit comments 👏

File: src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts [hunk 3]
```diff
@@ -259,4 +259,35 @@ describe('getNavItems', () => {
 		]
 	`)
 	})
+
+	it('for HCP, returns documentation nav link without dropdown', () => {
```
Comment (pbortnick) [inline src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts:263]: Added a test here to make sure the nav items are working for HCP @kendallstrautman

Comment (kendallstrautman) [reply src/components/navigation-header/components/product-page-content/utils/__tests__/get-nav-items.test.ts:263]: 💯 🙇‍♀️

**Review by kendallstrautman — APPROVED** [review]

**Pull request merged**